### DISCUSSION TITLE
Can't find conf file when installing as a docker container

### DISCUSSION THREAD

**Discussion by c0pper** [nicolargo/glances#2232] [General]
#### Description
I'm trying to setup the influxdb export in order to see stats in grafana.
I'm reading [this page of the documentation](https://glances.readthedocs.io/en/latest/gw/influxdb.html#influxdb-v2-from-influxdb-v1-8-x-flux-and-influxdb-v2-x) but I can't figure out where to find the  glances.conf when installing as a docker container. Here's my docker run command:
`docker run -d --name glances --restart="always" -p 61208-61209:61208-61209 -e GLANCES_OPT="-w" -v /var/run/docker.sock:/var/run/docker.sock:ro -v /home/pi/docker/configs/glances/glances.conf:/glances/conf/glances.conf --pid host nicolargo/glances:latest`
glances.conf on my host file system is an empty folder. Do i need to create the glances.conf file or what?
Thanks.

**Comment by elpigeondindo** [comment]
i see that glances start with the file where is /etc/glances.conf

so i put this :

- /home/pi/docker/configs/glances/glances.conf:/etc/glances.conf

after just copy a sample config in

**Comment by c0pper** [comment]
Thanks, that worked. However now I can't figure out how to export the data to influx db. I tried running 
`docker exec -it glances glances --export influxdb2`
but this error came up
```
Traceback (most recent call last):
  File "/usr/local/bin/glances", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 179, in main
    start(config=core.get_config(), args=core.get_args())
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 108, in start
    mode = GlancesMode(config=config, args=args)
  File "/usr/lib/python3.10/site-packages/glances/standalone.py", line 45, in __init__
    self.stats = GlancesStats(config=config, args=args)
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 39, in __init__
    self.load_modules(self.args)
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 90, in load_modules
    self.load_exports(args=args)
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 162, in load_exports
    export_module = __import__(self._exports_all[export_name])
  File "/usr/lib/python3.10/site-packages/glances/exports/glances_influxdb2.py", line 18, in <module>
    from influxdb_client import InfluxDBClient, WriteOptions
ModuleNotFoundError: No module named 'influxdb_client'
```

I have influxdb2 running in another docker container, pulled the latest image from here https://hub.docker.com/_/influxdb

**Comment by elpigeondindo** [comment]
i haved the same problem and i solved it like this. Try this

`docker exec -it glances glances --export influxdb2 --quiet`

To export to another influxdb docker you can use the config he created for that

```
version: "3.2"
services:
  glances:
    image: nicolargo/glances:dev
    restart: unless-stopped
    environment:
      # with influxdb and Grafana as interface
      - GLANCES_OPT=-q --export influxdb2 --time 10
    pid: "host"
    privileged: true
    network_mode: "host"
    depends_on:
      - influxdb
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./cfg/glances/glances.conf:/glances/conf/glances.conf

  influxdb:
    image: influxdb
    restart: unless-stopped
    ports:
      - 8086:8086
    env_file:
      - 'env.influxdb2'
    volumes:
      # Data persistency
      # sudo mkdir -p /srv/docker/influxdb2/data
      - /srv/docker/influxdb2-grafana/influxdb2/data:/var/lib/influxdb
      - ./cfg/influxdb2:/etc/influxdb2
    networks:
      - grafana_network

  grafana:
    image: grafana/grafana
    restart: unless-stopped
    ports:
      - 8080:3000
    user: "1000"
    volumes:
      - /srv/docker/influxdb2-grafana/grafana/data:/var/lib/grafana
    networks:
      - grafana_network

networks:
  grafana_network:
    # name: grafana_network
    driver: bridge
    driver_opts:
      com.docker.network.bridge.name: br_grafana
```

**Comment by RazCrimson** [comment]
> Thanks, that worked. However now I can't figure out how to export the data to influx db. I tried running `docker exec -it glances glances --export influxdb2` but this error came up
> 
> ```
> Traceback (most recent call last):
>   File "/usr/local/bin/glances", line 8, in <module>
>     sys.exit(main())
>   File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 179, in main
>     start(config=core.get_config(), args=core.get_args())
>   File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 108, in start
>     mode = GlancesMode(config=config, args=args)
>   File "/usr/lib/python3.10/site-packages/glances/standalone.py", line 45, in __init__
>     self.stats = GlancesStats(config=config, args=args)
>   File "/usr/lib/python3.10/site-packages/glances/stats.py", line 39, in __init__
>     self.load_modules(self.args)
>   File "/usr/lib/python3.10/site-packages/glances/stats.py", line 90, in load_modules
>     self.load_exports(args=args)
>   File "/usr/lib/python3.10/site-packages/glances/stats.py", line 162, in load_exports
>     export_module = __import__(self._exports_all[export_name])
>   File "/usr/lib/python3.10/site-packages/glances/exports/glances_influxdb2.py", line 18, in <module>
>     from influxdb_client import InfluxDBClient, WriteOptions
> ModuleNotFoundError: No module named 'influxdb_client'
> ```
> 
> I have influxdb2 running in another docker container, pulled the latest image from here https://hub.docker.com/_/influxdb

I think the issue is due to using the plain image. 

Try the docker images tagged with the `-full` suffix.

The plain images don't install the optional dependencies and influxdb_client is an optional dependency. The `-full` image should have all the dependencies installed.

-----

On an another note, this issue should be closed if there are no other problems.

Switching to a discussion would be better if you have more questions related to setting up an influx-db export.
